### PR TITLE
initial implementation of etcd-cpp-apiv3

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **etcd-cpp-apiv3/0.15.4**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
@SSE4 @uilianries 
I need some help from the community here.
Why absl includes are not found in grpc?
https://github.com/conan-io/conan-center-index/blob/be01e0c02ada74dcd20d77f8b25fc976f70a3175/recipes/grpc/all/conanfile.py#L98

It's declared with transitive headers...

**Comment by AndreyMlashkin** [comment]
@uilianries @SSE4 
I don't have these issues.
I have implemented deleting cmake files:
rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))

And I have also marked cpprestsdk as transitive

File: recipes/etcd-cpp-apiv3/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.libs = collect_libs(self)
```
Comment (uilianries) [inline recipes/etcd-cpp-apiv3/all/conanfile.py:76]:
```suggestion
        self.cpp_info.libs = ["etcd-cpp-api"]
```

Please, prefer using the library name always, so we have a hint to know what to expect.

**Review by uilianries — COMMENTED** [review]
@AndreyMlashkin Thank you for reporting that error. It's an outdated output, I tested your PR locally there is no such error. So, I just restart its build.

Please, always ping `@conan-io/barbarians` as described https://github.com/conan-io/conan-center-index/pull/22596#issuecomment-1926195474. It will ping Conan team members directly, instead only few.

**Comment by AndreyMlashkin** [comment]
Screenshot from 2024-02-06 09-44-01
Another CI bug?
It says, .github/workflows/stale.yml was changed, but it's not in the list of changed files

**Comment by uilianries** [comment]
@AndreyMlashkin Please, consider the PR https://github.com/AndreyMlashkin/conan-center-index/pull/11 as my review. It contains several suggestions related to this PR.

File: recipes/etcd-cpp-apiv3/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        env = VirtualBuildEnv(self)
+        env.generate()
+        if self._is_legacy_one_profile:
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
+
+        tc = CMakeToolchain(self)
+        tc.variables["gRPC_VERSION"] = self.dependencies["grpc"].ref.version
+        tc.variables["ETCD_CMAKE_CXX_STANDARD"] = self.settings.compiler.get_safe("cppstd", "11")
+        tc.variables["OpenSSL_DIR"] = self.dependencies["openssl"].package_folder
```
Comment (uilianries) [inline recipes/etcd-cpp-apiv3/all/conanfile.py:71]:
```suggestion
        tc.variables["OpenSSL_DIR"] = self.dependencies["openssl"].package_folder.replace('\\', '/')
```

Windows fails to parse `\`

**Comment by uilianries** [comment]
> ERROR: Missing prebuilt package for 'openssl/3.2.1'

I just started an internal CI build to generate the missing package.

**Comment by uilianries** [comment]
I just restarted the PR now. OpenSSL has been generated.

**Comment by uilianries** [comment]
Started another internal build to generate the missing `cpprestsdk/2.10.19`

**Comment by AndreyMlashkin** [comment]
> Started another internal build to generate the missing `cpprestsdk/2.10.19`

will the build automatically restarted or should I restart it after some time?

**Comment by uilianries** [comment]
> will the build automatically restarted or should I restart it after some time?

It's automated, don't worry. Yesterday, Jenkins was under maintenance as scheduled and discussed here: https://github.com/conan-io/conan-center-index/discussions/20497

This PR should be running soon.

**Comment by uilianries** [comment]
Hmm, the project uses abseil which requires C++11, but there is no standard configured. I gonna check locally.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

**Pull request merged**